### PR TITLE
Exit with children

### PR DESCRIPTION
**Pull request by sabrehagen**
When using the `--start-child` option, `--exit-with-children` should be used to exit the X server when the child process ends.

image

### CODE REVIEW

**Comment by sabrehagen** [comment]
Hmm, I notice you're using `xhost +x` as the command, which terminates immediately after execution. Presumably using the `--exit-with-children` option would cause the server to exit immediately. Maybe `start` should be used instead as per the warning.

**Comment by JAremko** [comment]
I think it simpler to synchronize starting Xpra and its child process with the current approach.

**Comment by JAremko** [comment]
I think it simpler to synchronize starting Xpra and its child process with the current approach. 

- - - - - - -

Actually we can replace it with `xpra start --start="$@"` Haven't tested the flag. But it should do the same.

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@ FROM alpine:edge
 MAINTAINER JAremko <w3techplaygound@gmail.com>
 
 # Kudos to @urzds for Xpra building example
-ENV XPRA_VERSION=2.3.2
+ENV XPRA_VERSION=2.5rc5
```
Comment (JAremko) [inline Dockerfile:6]: Usually Xpra version bump brings new dependencies. Have you tested it?
It will be better to wait for the release.

Comment (totaam) [reply Dockerfile:6]: xpra 2.5 has been out for over a week:
https://xpra.org/trac/wiki/News
IIRC: there are few new dependencies (python-xdg?) and some have been dropped (websockets modules).

Comment (JAremko) [reply Dockerfile:6]: @totaam  Thx! :heart: 

There are some minor issues:
For some reason it tests for `nvcc`  while all CUDA flags are off:
```
Xpra version 2.5
found pkg-config version: 1.6.1
build switches:
* PIC                  : Y
* Xdummy               : Y
* Xdummy_wrapper       : Y
* annotate             : Y
* bencode              : Y
* bundle_tests         : N
* client               : N
* clipboard            : Y
* csc_libyuv           : N
* csc_swscale          : Y
* cuda_kernels         : N
* cuda_rebuild         : N
* cython               : Y
* cython_bencode       : Y
* data                 : Y
* dbus                 : Y
* debug                : N
* dec_avcodec2         : N
* enc_ffmpeg           : Y
* enc_x264             : Y
* enc_x265             : N
* example              : Y
* gtk2                 : Y
* gtk3                 : N
* gtk_x11              : Y
* html5                : Y
* html5_brotli         : Y
* html5_gzip           : Y
* jpeg_decoder         : N
* jpeg_encoder         : N
* keyboard             : Y
* mdns                 : N
* minify               : Y
* modules              : Y
* netdev               : Y
* notifications        : Y
* nvenc                : N
* nvfbc                : N
* opengl               : N
* pam                  : N
* pillow               : Y
* printing             : N
* proxy                : Y
* rebuild              : Y
* rfb                  : Y
* scripts              : Y
* sd_listen            : N
* server               : Y
* service              : Y
* shadow               : Y
* sound                : N
* strict               : N
* tests                : N
* uinput               : N
* v4l2                 : Y
* verbose              : N
* vpx                  : Y
* vsock                : Y
* warn                 : Y
* webcam               : N
* webp                 : N
* x11                  : Y
* xdg_open             : Y
* xinput               : Y
'nvcc --version' failed with return code 127
stderr: None
updated ./xpra/build_info.py with:
* BUILD_BIT            = 64bit
* BUILD_CPU            = Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz
* BUILD_DATE           = 2019-03-31
* BUILD_MACHINE        = x86_64
* BUILD_OS             = Linux   
* BUILD_TIME           = 21:51
* BUILT_BY             = root
* BUILT_ON             = cdd3702a3247
* COMPILER_VERSION     = gcc (Alpine 8.3.0) 8.3.0
* CYTHON_VERSION       = 0.29.2
* LINKER_VERSION       = GNU ld (GNU Binutils) 2.31.1
* PYTHON_VERSION       = 2.7.15
* RELEASE_BUILD        = True
* lib_pygobject        = 2.28.7
* lib_pygtk            = 2.24.0
* lib_vpx              = 1.8.0
* lib_x11              = 1.6.7
* lib_x264             = 0.152.x
* lib_xcomposite       = 0.4.4
* lib_xdamage          = 1.1.4
* lib_xext             = 1.3.3
* lib_xfixes           = 5.0.3
* lib_xkbfile          = 1.0.9
* lib_xrandr           = 1.5.1
* lib_xtst             = 1.2.3
```

Minification fails with latest `uglify-js` (had to use `uglify-js@2`):

```
minifying html5 client to '/usr/share/xpra/www' using uglifyjs
Error: failed to minify 'js/lib/broadway/Decoder.js', command returned error 1
Error: failed to minify 'js/Protocol.js', command returned error 1
Error: failed to minify 'js/Window.js', command returned error 1
Error: failed to minify 'js/Keycodes.js', command returned error 1
Error: failed to minify 'js/MediaSourceUtil.js', command returned error 1
Error: failed to minify 'js/Utilities.js', command returned error 1
Error: failed to minify 'js/Client.js', command returned error 1
Error: failed to minify 'js/Notifications.js', command returned error 1
Error: failed to minify 'js/lib/aurora/aurora.js', command returned error 1
Error: failed to minify 'js/lib/aurora/flac.js', command returned error 1
Error: failed to minify 'js/lib/aurora/mp3.js', command returned error 1
Error: failed to minify 'js/lib/aurora/aurora-xpra.js', command returned error 1
Error: failed to minify 'js/lib/aurora/aac.js', command returned error 1
Error: failed to minify 'js/lib/wsworker_check.js', command returned error 1
Error: failed to minify 'js/lib/jquery.ba-throttle-debounce.js', command returned error 1
Error: failed to minify 'js/lib/jsmpeg.js', command returned error 1
Error: failed to minify 'js/lib/zlib.js', command returned error 1
Error: failed to minify 'js/lib/jszip.js', command returned error 1
Error: failed to minify 'js/lib/lz4.js', command returned error 1
Error: failed to minify 'js/lib/jquery-ui.js', command returned error 1
Error: failed to minify 'js/lib/bencode.js', command returned error 1
Error: failed to minify 'js/lib/FileSaver.js', command returned error 1
Error: failed to minify 'js/lib/jquery.js', command returned error 1
Error: failed to minify 'js/lib/es6-shim.js', command returned error 1
Error: failed to minify 'js/lib/forge.js', command returned error 1
```

Xpra compiles and works (tested only HTML5 thingy)

**Comment by JAremko** [comment]
@sabrehagen  I updated Xpra to the latest one and decided to stick for now with the old calling convention (i.e. keep `-exit-with-children`) but thanks for suggestion anyway :+1:

File: Dockerfile (hunk 1 above)
Comment (JAremko) [reply Dockerfile:6]: @totaam Yeah also curl failed to validate `www.xpra.org` certificate so I switched to `http` + checksum.

**Comment by totaam** [comment]
@JAremko 

- I am unable to reproduce the nvcc warnings: the whole nvcc section of the setup.py file is guarded by a `if nvenc_ENABLED and cuda_kernels_ENABLED:`. If you can reproduce it, please place `assert False` as the first line of code in `get_nvcc_version` so we can see how it got there.
- As for uglifyjs, thanks for the heads up, I'll try to make it compatible with the newer version once it lands in one of the supported distros
- the https validation should work, we use a let's encrypt cert, renewed again just days ago

**Comment by JAremko** [comment]
@totaam  here the full build log (from DockerHub) https://gist.github.com/JAremko/7db980a70ae7626ad345a84ea870aace#file-build-log-L530 

Just in case if it can help with something. :smile:

**Comment by JAremko** [comment]
> the https validation should work, we use a let's encrypt cert, renewed again just days ago

@totaam  try: `curl https://www.xpra.org/src/xpra-2.5.tar.xz`

It gives me:

> curl: (60) SSL: no alternative certificate subject name matches target host name
 'www.xpra.org'
More details here: https://curl.haxx.se/docs/sslcerts.html
curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

**Comment by totaam** [comment]
> here the full build log

Sadly, it doesn't.

> @totaam try: curl https://www.xpra.org/src/xpra-2.5.tar.xz

Just remove the `www.` or try again, I've renewed the cert with an alias for `www.xpra.org`.

**Pull request closed without merging**